drivers/periph/i2c: Updated i2c_release() to return void

maribu opened the pull request:
### Contribution description

`i2c_release()` should not return any value, as:

- There is no reasonable error handling possible by the caller, so there is no value in indicating success/failure via the return value to the caller
- There is no legitimate reason to fail *unless* an invalid I2C bus was released or an I2C bus that was not previously acquired was released
    - --> This would indicate a bug in the code and should be tackled by an `assert()`

### Testing procedure

1. Murdock will do the most important part
2. Test if the I2C interface is still working

### Issues/PRs references

https://github.com/RIOT-OS/RIOT/issues/10673

benpicco commented:
tests/periph_i2c needs to be updated as well.

Also please add an additional commit to add parenthesis to [`cc2538/periph/i2c.c`](https://github.com/RIOT-OS/RIOT/blob/69179349467fbdc9f28e101fb1bff41e4b849023/cpu/cc2538/periph/i2c.c#L162) so Murdock & cppcheck are happy.

maribu commented:
As now one could have potentially looked into the last two commits, I fixed an issue and squashed those two directly. (The commits older than 13 minutes have not been changed, as github correctly shows.)

kaspar030 commented on cpu/cc2538/periph/i2c.c:
```diff
@@ -159,7 +159,7 @@ static void _i2c_master_slave_addr(uint16_t addr, bool receive)
     DEBUG("%s (%" PRIx16 ", %d)\n", __FUNCTION__, addr, (int)receive);
     assert(!(addr & 0x80));
-    I2CM_SA = (addr << 1) | receive;
+    I2CM_SA = (addr << 1) | (receive ? 0x1 : 0x0);
```
receive is 0 or 1 anyways?

benpicco replied:
`cppcheck` complains otherwise.
Maybe casting it to `(int)` would be a better solution?

maribu replied:
cppcheck was unhappy with an integer type and `bool` being in a bitwise operation. I could have explicitly casted the bool to the type of `I2CM_SA`. But translating it to an integer literal with the same value has the advantage, that this will continue to work when the type of `I2CM_SA` is every changed in the future.

I expect the compiler to generate the exact same machine code.

maribu replied:
> Maybe casting it to `(int)` would be a better solution?

`cpu/cc2538/include/cc2538.h:207`:
```C
#define I2CM_SA                     ( *(cc2538_reg_t*)0x40020000 ) /**< I2C Master Slave address */
```

`cpu/cc2538/include/cc2538.h:124`:
```C
typedef volatile uint32_t cc2538_reg_t; /**< Least-significant 32 bits of the IEEE address */
```

So casting to `(int)` would result in different sign. Casting to `cc2538_reg_t` would add the `volatile` qualifier. The use of `volatile` for the actual register `I2CM_SA` is correct, but for the parameter `receive` would result int he code to be less efficient. So if casting is preferred, it should be casted to `uint32_t`.

kaspar030 replied:
well, if this makes cppcheck happy, I guess we can keep it. Probably there are even people who find it more readable. ;)

dylad commented:
Is this PR affect by the soft feature freeze ?

benpicco commented:
I think this is a minor change.

maribu commented:
@gschorcht: Mind to provide the second ACK? This way 50% of your goal to remove return values from `i2c_acquire()` and `i2c_release()` would be reached ;-)
